src/context.hh

```diff
@@ -28,16 +28,14 @@ struct JumpList
 
     friend bool operator==(const JumpList& lhs, const JumpList& rhs)
     {
-        return lhs.m_jumps == rhs.m_jumps and lhs.m_current == rhs.m_current;
+        return lhs == rhs and lhs.m_current == rhs.m_current;
```

Comment:
This looks like its going to be infinitely recursive, you'd need to cast to the base class or call vector::operator== explicitely, but either way I think it shows inheriting from std::vector might not be that nice here.